server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/SessionRoutesContract.scala

```diff
@@ -171,4 +171,111 @@ trait SessionRoutesContract {
 
     assertThatJson(sessionJson).isEqualTo(expected_session_object)
   }
+
+  @Test
+  def getResponseShouldReturnUrlEndpointContainJmapPrefixWhenAssignHeaderAndConfigurationProvided(): Unit = {
+    val sessionJson: String = `given`()
+    .when()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .header("X-JMAP-PREFIX", "/oidc")
```

Comment:
IMO we should have here a fully qualified URL. This gives greater flexibility when configuring an API gateway.

Reply:
What do you think about this?
https://github.com/apache/james-project/pull/780#discussion_r763726881

Reply:
I think the two header approach is valid.